### PR TITLE
county field dropdown

### LINKED ISSUE #2771: Optionally restrict county field to a dropdown
Thanks for creating an issue! Please fill out this form so we can be sure to have all the information we need, and to minimize back and forth.

* **What are we trying to do?**
Some funds work only in service areas, and so geo matters a lot to them. We enumerate states since there aren't that many valid ones, but sometimes there are only so many counties funds care about, so it might make sense to enumerate these as well with a dropdown rather than just the free text input field.

* **What feature or behavior is this required for?**
cleaner patient info tracking

* **How could we solve this issue? (Not knowing is okay!)**
This is probably a config of comma separated values where each value is a county, and if the config is set this means that you get a dropdown something like `[nil, values, ..., other]` instead of the free text field. (My hunch is that we don't need to support free text Other values, and in fact that would be a bit of a hassle.)

* **Anything else?**
nah

### PR DESCRIPTION
**Pull request by elimbaum**
I rule and have completed some work on Case Manager that's ready for review!

(brief, plain english overview of your changes here)

This pull request makes the following changes:
* adds a config for county field
* if config is defined, display a dropdown
* otherwise, display text field (existing behavior)

(If there are changes to the views, please include a screenshot so we know what to look for!)

It relates to the following issue #s: 
* Fixes #2771 

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

**Comment by elimbaum** [comment]
Screen Shot 2023-01-25 at 20 50 00

example: prior value was `Test`; dropdown options `A B C` populate.

File: app/views/patients/_patient_information.html.erb
Comment (elimbaum) [inline app/views/patients/_patient_information.html.erb:49]: --> new view code

Comment (colinxfleming) [reply app/views/patients/_patient_information.html.erb:49]: well handled I think! maybe a system test confirming this works, but I don't think that's completely necessary and defer to you.

File: app/helpers/patients_helper.rb
Comment (colinxfleming) [inline app/helpers/patients_helper.rb:120]: could we get a quick unit test for this?

**Review by colinxfleming — APPROVED** [review]
great work as usual! I think some additional quick tests (unit test for county_options dropdown, maybe a system test testing that it swaps to the dropdown if the config is set) would be good here but I don't think they're critical - interested in whether you think they're worth it.

Once we have that square we can merge, but this is in mergeable shape as is, so marking as such!

File: db/schema.rb
Comment (colinxfleming) [inline db/schema.rb:51]: these have gone back and forth several times and I'm choosing not to think about it very hard anymore

**Review by colinxfleming — APPROVED** [review]
outstanding work as per usual! nothing to spruce up here, thanks much @elimbaum !

**Pull request merged**
